# pediatric-AML-ECM-transcriptomics
# Create a reproducible analysis workflow R script for the pediatric AML ECM study

script_lines <- c(
  "# pediatric_AML_ECM_reproducible_workflow.R",
  "# Reproducible analysis workflow for: Integrative pediatric AML transcriptomics reveals ECM remodeling as a therapeutic vulnerability",
  "# This script assumes that raw counts / expression matrices and sample metadata are stored in a local 'data/' directory",
  "# and that a configuration file (config.yml or config.csv) specifies sample grouping and contrasts.",
  "",
  "###############################################",
  "# 1. Setup & package loading",
  "###############################################",
  "suppressPackageStartupMessages({",
  "  library(tidyverse)",
  "  library(DESeq2)",
  "  library(edgeR)",
  "  library(limma)",
  "  library(clusterProfiler)",
  "  library(org.Hs.eg.db)",
  "  library(enrichplot)",
  "  library(msigdbr)",
  "  library(STRINGdb)",
  "  library(igraph)",
  "  library(decoupleR)",
  "  library(dorothea)",
  "})",
  "",
  "set.seed(1234)",
  "",
  "###############################################",
  "# 2. Helper functions",
  "###############################################",
  "run_deseq2 <- function(counts, coldata, design_formula) {",
  "  dds <- DESeqDataSetFromMatrix(countData = counts,",
  "                                  colData = coldata,",
  "                                  design = design_formula)",
  "  dds <- dds[rowSums(counts(dds)) > 10, ]  # basic filter",
  "  dds <- DESeq(dds)",
  "  res <- results(dds)",
  "  res_df <- as.data.frame(res) %>%",
  "    rownames_to_column(var = 'gene') %>%",
  "    arrange(padj)",
  "  list(dds = dds, res = res_df)",
  "}",
  "",
  "run_limma_voom <- function(counts, group) {",
  "  dge <- DGEList(counts = counts)",
  "  keep <- filterByExpr(dge, group = group)",
  "  dge <- dge[keep, , keep.lib.sizes = FALSE]",
  "  dge <- calcNormFactors(dge)",
  "  design <- model.matrix(~ 0 + group)",
  "  v <- voom(dge, design, plot = FALSE)",
  "  fit <- lmFit(v, design)",
  "  # User should specify contrasts outside this helper",
  "  list(v = v, fit = fit, design = design)",
  "}",
  "",
  "run_go_enrichment <- function(genes, universe = NULL, ont = 'BP') {",
  "  enrichGO(gene          = genes,",
  "          OrgDb         = org.Hs.eg.db,",
  "          keyType       = 'SYMBOL',",
  "          universe      = universe,",
  "          ont           = ont,",
  "          pAdjustMethod = 'BH',",
  "          pvalueCutoff  = 0.05,",
  "          qvalueCutoff  = 0.2)",
  "}",
  "",
  "run_kegg_enrichment <- function(genes_entrez, universe = NULL) {",
  "  enrichKEGG(gene          = genes_entrez,",
  "             organism      = 'hsa',",
  "             universe      = universe,",
  "             pAdjustMethod = 'BH',",
  "             pvalueCutoff  = 0.05,",
  "             qvalueCutoff  = 0.2)",
  "}",
  "",
  "run_gsea_msigdb <- function(ranked_genes, category = 'H', subcategory = NULL) {",
  "  msig <- msigdbr(species = 'Homo sapiens', category = category, subcategory = subcategory)",
  "  gsea_list <- ranked_genes",
  "  gsea_res <- GSEA(gsea_list, TERM2GENE = msig %>% dplyr::select(gs_name, gene_symbol))",
  "  gsea_res",
  "}",
  "",
  "###############################################",
  "# 3. Data loading (user to adapt paths/filenames)",
  "###############################################",
  "# Example expected files (user should adapt to actual filenames):",
  "# data/TARGET_AML_counts.csv",
  "# data/TARGET_AML_coldata.csv",
  "# data/GSE246783_counts_MI3454.csv",
  "# data/GSE292324_counts_PRMT5.csv",
  "# data/GSE292050_counts_NID1.csv",
  "# data/TCGA_AML_expression.csv",
  "# data/GTEx_blood_expression.csv",
  "",
  "# TARGET-AML (pediatric)",
  "target_counts <- read.csv('data/TARGET_AML_counts.csv', row.names = 1, check.names = FALSE)",
  "target_coldata <- read.csv('data/TARGET_AML_coldata.csv', row.names = 1)",
  "",
  "# MI3454 (GSE246783)",
  "mi3454_counts <- read.csv('data/GSE246783_counts_MI3454.csv', row.names = 1, check.names = FALSE)",
  "mi3454_coldata <- read.csv('data/GSE246783_coldata_MI3454.csv', row.names = 1)",
  "",
  "# PRMT5 inhibition (GSE292324)",
  "prmt5_counts <- read.csv('data/GSE292324_counts_PRMT5.csv', row.names = 1, check.names = FALSE)",
  "prmt5_coldata <- read.csv('data/GSE292324_coldata_PRMT5.csv', row.names = 1)",
  "",
  "# NID1 knockdown (GSE292050)",
  "nid1_counts <- read.csv('data/GSE292050_counts_NID1.csv', row.names = 1, check.names = FALSE)",
  "nid1_coldata <- read.csv('data/GSE292050_coldata_NID1.csv', row.names = 1)",
  "",
  "###############################################",
  "# 4. Differential expression analyses",
  "###############################################",
  "# MI3454: treatment vs control",
  "mi3454_coldata$group <- factor(mi3454_coldata$group, levels = c('control', 'MI3454'))",
  "mi3454_res <- run_deseq2(mi3454_counts, mi3454_coldata, ~ group)",
  "write.csv(mi3454_res$res, file = 'results/DEG_MI3454_DESeq2.csv', row.names = FALSE)",
  "",
  "# PRMT5 inhibition: inhibitor vs vehicle",
  "prmt5_coldata$group <- factor(prmt5_coldata$group, levels = c('control', 'PRMT5_inhib'))",
  "prmt5_res <- run_deseq2(prmt5_counts, prmt5_coldata, ~ group)",
  "write.csv(prmt5_res$res, file = 'results/DEG_PRMT5_DESeq2.csv', row.names = FALSE)",
  "",
  "# NID1 knockdown: shNID1 vs shControl",
  "nid1_coldata$group <- factor(nid1_coldata$group, levels = c('control', 'NID1_KD'))",
  "nid1_res <- run_deseq2(nid1_counts, nid1_coldata, ~ group)",
  "write.csv(nid1_res$res, file = 'results/DEG_NID1_DESeq2.csv', row.names = FALSE)",
  "",
  "###############################################",
  "# 5. Functional enrichment & pathway analysis",
  "###############################################",
  "dir.create('results', showWarnings = FALSE)",
  "",
  "# Example: upregulated genes in MI3454 (padj < 0.05, log2FC > 0.58)",
  "mi3454_deg <- mi3454_res$res %>%",
  "  filter(!is.na(padj)) %>%",
  "  filter(padj < 0.05, log2FoldChange > 0.58)",
  "",
  "mi3454_up_genes <- mi3454_deg$gene",
  "mi3454_go_bp <- run_go_enrichment(mi3454_up_genes, ont = 'BP')",
  "mi3454_go_mf <- run_go_enrichment(mi3454_up_genes, ont = 'MF')",
  "mi3454_go_cc <- run_go_enrichment(mi3454_up_genes, ont = 'CC')",
  "",
  "saveRDS(mi3454_go_bp, file = 'results/MI3454_GO_BP.rds')",
  "saveRDS(mi3454_go_mf, file = 'results/MI3454_GO_MF.rds')",
  "saveRDS(mi3454_go_cc, file = 'results/MI3454_GO_CC.rds')",
  "",
  "###############################################",
  "# 6. GSEA using ranked gene lists",
  "###############################################",
  "mi3454_ranked <- mi3454_res$res %>%",
  "  filter(!is.na(stat)) %>%",
  "  arrange(desc(stat)) %>%",
  "  select(gene, stat)",
  "",
  "mi3454_ranked_vec <- mi3454_ranked$stat",
  "names(mi3454_ranked_vec) <- mi3454_ranked$gene",
  "",
  "mi3454_gsea_h <- run_gsea_msigdb(mi3454_ranked_vec, category = 'H')",
  "saveRDS(mi3454_gsea_h, file = 'results/MI3454_GSEA_Hallmark.rds')",
  "",
  "###############################################",
  "# 7. PPI network construction (STRING)",
  "###############################################",
  "# Example: build STRING network for MI3454 upregulated genes",
  "string_db <- STRINGdb$new(version = '12', species = 9606, score_threshold = 400, input_directory = '')",
  "mi3454_mapped <- string_db$map(data.frame(gene = mi3454_up_genes), 'gene', removeUnmappedRows = TRUE)",
  "mi3454_string_graph <- string_db$get_subnetwork(mi3454_mapped$STRING_id)",
  "saveRDS(mi3454_string_graph, file = 'results/MI3454_STRING_graph.rds')",
  "",
  "###############################################",
  "# 8. TF / upstream regulator inference (DoRothEA + decoupleR)",
  "###############################################",
  "data(dorothea_hs, package = 'dorothea')",
  "regulon <- dorothea_hs %>% filter(confidence %in% c('A', 'B', 'C'))",
  "",
  "# Example: compute TF activity on MI3454 variance-stabilized expression",
  "mi3454_vst <- vst(mi3454_res$dds, blind = TRUE)",
  "mi3454_vst_mat <- assay(mi3454_vst)",
  "",
  "tf_acts <- run_viper(mat = mi3454_vst_mat, regulon = regulon, nes = TRUE)",
  "saveRDS(tf_acts, file = 'results/MI3454_TF_activity_dorothea_decoupleR.rds')",
  "",
  "###############################################",
  "# 9. Session info for reproducibility",
  "###############################################",
  "writeLines(capture.output(sessionInfo()), con = 'results/sessionInfo.txt')",
  "",
  "message('Analysis workflow completed (example pipeline). Adapt file paths and group labels to your local data to fully reproduce the study.')"
)

writeLines(script_lines, 'pediatric_AML_ECM_reproducible_workflow.R')

print('Created pediatric_AML_ECM_reproducible_workflow.R in the current directory.')